### PR TITLE
Proxy

### PR DESCRIPTION
**Pull request by ob1vanq**
Add proxy param to login session

### CODE REVIEW

**Review by Natgho — COMMENTED** [review]
It seems perfect, it should be accepted as soon as possible! :)

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 __pycache__/
 .mypy_cache/
+.idea/
```
Comment (Thammus) [inline .gitignore:3]: Please remove this line. This does not belong in the `.gitignore` file.

File: instaloader/instaloadercontext.py [hunk 2]
```diff
@@ -266,7 +273,7 @@ def login(self, user, passwd):
         self._session = session
         self.username = user
 
-    def two_factor_login(self, two_factor_code):
+    def two_factor_login(self, two_factor_code, proxies):
```
Comment (Thammus) [inline instaloader/instaloadercontext.py:276]: The parameter `proxies` is not used.

File: instaloader/instaloader.py [hunk 3]
```diff
@@ -232,11 +232,12 @@ def __init__(self,
                  fatal_status_codes: Optional[List[int]] = None,
                  iphone_support: bool = True,
                  title_pattern: Optional[str] = None,
-                 sanitize_paths: bool = False):
+                 sanitize_paths: bool = False,
+                 proxies: dict = None):
```
Comment (Thammus) [inline instaloader/instaloader.py:236]: If the value can be `None` the type should reflect this through `Optional[Dict[str,str]]`.

**Review by Thammus — COMMENTED** [review]
This still lacks documentation.

Also the changes should go into the branch `upcoming/v4.10` instead of `master`.

**Comment by Natgho** [comment]
@ob1vanq are you planning the complete this PR? if you don't have to time, I can do it.

**Comment by ainyava** [comment]
I would really appreciate if the suitable proxy support would be added, Thanks :+1:

**Comment by LuckyCoders** [comment]
i use 
`export HTTPS_PROXY="https://113.53.60.178:8080"`

i get
```
Session file does not exist yet - Logging in.
Traceback (most recent call last):
  File "/usr/local/bin/instaloader", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/dist-packages/instaloader/__main__.py", line 471, in main
    _main(loader,
  File "/usr/local/lib/python3.9/dist-packages/instaloader/__main__.py", line 110, in _main
    instaloader.login(username, password)
  File "/usr/local/lib/python3.9/dist-packages/instaloader/instaloader.py", line 634, in login
    self.context.login(user, passwd)
  File "/usr/local/lib/python3.9/dist-packages/instaloader/instaloadercontext.py", line 214, in login
    csrf_json = self.get_json('accounts/login/', {}, session=session)
  File "/usr/local/lib/python3.9/dist-packages/instaloader/instaloadercontext.py", line 325, in get_json
    resp = sess.get('https://{0}/{1}'.format(host, path), params=params, allow_redirects=False)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 555, in get
    return self.request('GET', url, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 696, in urlopen
    self._prepare_proxy(conn)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 966, in _prepare_proxy
    conn.connect()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 359, in connect
    conn = self._connect_tls_proxy(hostname, conn)
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 500, in _connect_tls_proxy
    return ssl_wrap_socket(
  File "/usr/lib/python3/dist-packages/urllib3/util/ssl_.py", line 453, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls)
  File "/usr/lib/python3/dist-packages/urllib3/util/ssl_.py", line 495, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock)
  File "/usr/lib/python3.9/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.9/ssl.py", line 997, in _create
    raise ValueError("check_hostname requires server_hostname")
ValueError: check_hostname requires server_hostname

```

**Pull request closed without merging**